cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -166,45 +180,72 @@ func (o *objectGraph) objToNode(obj *unstructured.Unstructured) *node {
 		tenantClusters: make(map[*node]empty),
 		tenantCRSs:     make(map[*node]empty),
 		virtual:        false,
+		forceMove:      o.getForceMove(obj.GetKind(), obj.GetAPIVersion()),
 	}
 
 	o.uidToNode[newNode.identity.UID] = newNode
 	return newNode
 }
 
+func (o *objectGraph) getForceMove(kind string, apiVersion string) bool {
+	kindAPIStr := getKindAPIString(metav1.TypeMeta{Kind: kind, APIVersion: apiVersion})
+	if _, ok := o.types[kindAPIStr]; ok {
+		if o.types[kindAPIStr].forceMove {
+			return true
+		}
+	}
+	return false
+}
+
 // getDiscoveryTypes returns the list of TypeMeta to be considered for the the move discovery phase.
 // This list includes all the types defines by the CRDs installed by clusterctl and the ConfigMap/Secret core types.
-func (o *objectGraph) getDiscoveryTypes() ([]metav1.TypeMeta, error) {
-	discoveredTypes := []metav1.TypeMeta{}
-
+func (o *objectGraph) getDiscoveryTypes() error {
 	crdList := &apiextensionsv1.CustomResourceDefinitionList{}
 	getDiscoveryTypesBackoff := newReadBackoff()
 	if err := retryWithExponentialBackoff(getDiscoveryTypesBackoff, func() error {
 		return getCRDList(o.proxy, crdList)
 	}); err != nil {
-		return nil, err
+		return err
 	}
 
+	o.types = make(map[string]*discoveryTypeInfo)
+
 	for _, crd := range crdList.Items {
 		for _, version := range crd.Spec.Versions {
 			if !version.Storage {
 				continue
 			}
 
-			discoveredTypes = append(discoveredTypes, metav1.TypeMeta{
+			typeMeta := metav1.TypeMeta{
 				Kind: crd.Spec.Names.Kind,
 				APIVersion: metav1.GroupVersion{
 					Group:   crd.Spec.Group,
 					Version: version.Name,
 				}.String(),
-			})
+			}
+
+			o.types[getKindAPIString(typeMeta)] = &discoveryTypeInfo{typeMeta: typeMeta}
+			if _, ok := crd.Labels[clusterctlv1.ClusterctlMoveLabelName]; ok {
+				o.types[getKindAPIString(typeMeta)].forceMove = true
+			}
```

Comment:
nit
move this above L227 and change it into
```suggestion
			if _, ok := crd.Labels[clusterctlv1.ClusterctlMoveLabelName]; ok {
				typeMeta.forceMove = true
			}
```